File: iris-ld/iris-ld-jsonld/src/main/java/city/smartb/iris/jsonld/jackson/ObjectMapperFactory.java
package city.smartb.iris.jsonld.jackson;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.json.JsonMapper;
import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;

public class ObjectMapperFactory {

    private static final ObjectMapper mapper;

    static {
        mapper = JsonMapper.builder()
                .addModule(new Jdk8Module())
                .addModule(new JavaTimeModule())
                .build();
    }

    public ObjectMapper create() {
        return mapper;
    }
}

File: iris-ld/iris-ld-jsonld/src/main/kotlin/city/smartb/iris/ld/jsonld/jackson/ObjectMapperFactory.kt
package city.smartb.iris.ld.jsonld.jackson

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.json.JsonMapper
import com.fasterxml.jackson.datatype.jdk8.Jdk8Module
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule

class ObjectMapperFactory {
    fun create(): ObjectMapper {
        return mapper
    }

    companion object {
        private val mapper: ObjectMapper = JsonMapper.builder()
            .addModule(Jdk8Module())
            .addModule(JavaTimeModule())
            .build()

    }
}
